examples/historic_async.py

```diff
@@ -0,0 +1,126 @@
+        raise Exception('asyncio is not support in your python version')
+    print(f"Getting data for {len(symbols)} symbols, timeframe: {timeframe},"
+          f" between dates: start={start}, end={end}")
+
+    tasks = []
+
+    for symbol in symbols:
+        tasks.append(
+            rest.get_bars_async(symbol, start, end, timeframe, limit=500))
+    if minor >= 8:
+        results = await asyncio.gather(*tasks)
+    else:
+        results = await gather_with_concurrency(500, *tasks)
+
+    bad_requests = 0
+    for response in results:
+        if not len(response):
+            bad_requests += 1
+
+    print(f"Total of {len(results)} Bars, and {bad_requests} empty responses.")
+
+
+async def get_historic_trades(symbols, start, end, timeframe: TimeFrame):
+    major = sys.version_info.major
+    minor = sys.version_info.minor
+    if major < 3 or minor < 6:
+        raise Exception('asyncio is not support in your python version')
+    print(f"Getting data for {len(symbols)} symbols, timeframe: {timeframe},"
+          f" between dates: start={start}, end={end}")
+
+    tasks = []
+
+    for symbol in symbols:
+        tasks.append(
+            rest.get_trades_async(symbol, start, end, timeframe, limit=500))
+    if minor >= 8:
+        results = await asyncio.gather(*tasks)
+    else:
+        results = await gather_with_concurrency(500, *tasks)
+
+    bad_requests = 0
+    for response in results:
+        if not len(response):
+            bad_requests += 1
+
+    print(
+        f"Total of {len(results)} Trades, and {bad_requests} empty responses.")
+
+
+async def get_historic_quotes(symbols, start, end, timeframe: TimeFrame):
+    major = sys.version_info.major
+    minor = sys.version_info.minor
+    if major < 3 or minor < 6:
+        raise Exception('asyncio is not support in your python version')
+    print(f"Getting data for {len(symbols)} symbols, timeframe: {timeframe},"
+          f" between dates: start={start}, end={end}")
+
+    tasks = []
+
+    for symbol in symbols:
+        tasks.append(
+            rest.get_quotes_async(symbol, start, end, timeframe, limit=500))
+    if minor >= 8:
+        results = await asyncio.gather(*tasks)
+    else:
+        results = await gather_with_concurrency(500, *tasks)
+
+    bad_requests = 0
+    for response in results:
+        if not len(response):
+            bad_requests += 1
+
+    print(
+        f"Total of {len(results)} Quotes, and {bad_requests} empty responses.")
+
+
+async def main(symbols):
+    start = pd.Timestamp('2021-05-01', tz=NY).date().isoformat()
+    end = pd.Timestamp('2021-08-30', tz=NY).date().isoformat()
+    timeframe: TimeFrame = TimeFrame.Day
+    await get_historic_bars(symbols, start, end, timeframe)
+    await get_historic_trades(symbols, start, end, timeframe)
+    await get_historic_quotes(symbols, start, end, timeframe)
+
+
+if __name__ == '__main__':
+    import time
+    import yaml
+
+    with open("./config.yaml", mode='r') as f:
```

Comment:
An example should be given for this.

Reply:
added

Reply:
Where? I don't see it.

Reply:
Also, I think defaulting to the env vars would be nice in this example as well.

Reply:
in the docstring of the function
![image](https://user-images.githubusercontent.com/78740792/132317442-0c20f392-aa51-4f69-ae54-0ae438e008f5.png)

Reply:
now also checking the env variables
![image](https://user-images.githubusercontent.com/78740792/132317525-78458aaf-23c3-4b22-b131-3a04521e183d.png)